### PR TITLE
notmuch: fix Ruby option

### PR DESCRIPTION
**Pull request by DomT4**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

Closes https://github.com/Homebrew/homebrew-core/issues/30019.

### CODE REVIEW

**Comment by DomT4** [comment]
> looks like a bug that needs to be reported since we set --prefix=#{prefix}

Upstream throws out to Ruby to do the heavy lifting here via `ruby extconf.rb --vendor`, so the problematic parts here are being set by Ruby rather than upstream. Whether or not they should be is maybe a different question, but I don't think it's a bug so much as _a feature_ 🙃.

File: Formula/notmuch.rb [hunk 1]
```diff
@@ -56,6 +56,15 @@ def install
     system "./configure", *args
     system "make", "V=1", "install"
 
+    if build.with? "ruby"
+      cd "bindings/ruby" do
+        # Prevent Makefile from trying to break free of the
+        # sandbox and mkdir in HOMEBREW_PREFIX.
+        inreplace "Makefile", "#{HOMEBREW_PREFIX}/lib", lib
```
[the remark this answers is not in the document]
Comment (DomT4) [reply Formula/notmuch.rb]: Sure, in this case the `ruby` formula as defined by the dependency, and the `ruby` formula is configured to use `HOMEBREW_PREFIX`, so whatever `HOMEBREW_PREFIX` is should always be the needed `inreplace` here, unless I'm missing something obvious? I've been awake a literal day at this point so it's possible.

[the remark this answers is not in the document]
Comment (ilovezfs) [reply Formula/notmuch.rb]: Sorry I have vendor ruby on the brain so misunderstood "Homebrew's `ruby`".

[the remark this answers is not in the document]
Comment (ilovezfs) [reply Formula/notmuch.rb]:
```
bash-3.2$ grep "usr\/local\/lib" Makefile 
vendordir = $(DESTDIR)/usr/local/lib/ruby/vendor_ruby
sitedir = $(DESTDIR)/usr/local/lib/ruby/site_ruby
ldflags  = -L. -fstack-protector -L/usr/local/lib -L/usr/local/opt/libyaml/lib -L/usr/local/opt/openssl/lib -L/usr/local/opt/readline/lib 
bash-3.2$ 
```
Just inreplacing that `-L/usr/local/lib` is probably not ideal.

[the remark this answers is not in the document]
Comment (ilovezfs) [reply Formula/notmuch.rb]: maybe replace `HOMEBREW_PREFIX/"lib/ruby/"`?

[the remark this answers is not in the document]
Comment (DomT4) [reply Formula/notmuch.rb]: No worries, and fair comment, I'll update that shortly.

**Pull request merged**